Domains/4-Application/Connectors/CivilInfrastructureFramework/CifRail.ecschema.xml

```diff
@@ -3,20 +3,37 @@
 |  * See LICENSE.md in the project root for license terms and full copyright notice.
 ======================================================================================= -->
-<ECSchema schemaName="CifRail" alias="cifrl" version="01.00.07" description="iModel Connector schema containing aspect classes with properties from OpenRail Designer." xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
-    <ECSchemaReference name="BisCore" version="01.00.14" alias="bis"/>
+<ECSchema schemaName="CifRail" alias="cifrl" version="01.00.08" description="iModel Connector schema containing aspect classes with properties from OpenRail Designer." xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+    <ECSchemaReference name="BisCore" version="01.00.16" alias="bis"/>
     <ECSchemaReference name="BisCustomAttributes" version="01.00.00" alias="bisCA"/>
-    <ECSchemaReference name="CifCommon" version="01.00.03" alias="cifcmn"/>
-    <ECSchemaReference name="CifUnits" version="01.00.01" alias="cifu"/>
-    <ECSchemaReference name="CoreCustomAttributes" version="01.00.03" alias="CoreCA"/>
+    <ECSchemaReference name="CifCommon" version="01.00.06" alias="cifcmn"/>
+    <ECSchemaReference name="CifUnits" version="01.00.03" alias="cifu"/>
+    <ECSchemaReference name="CoreCustomAttributes" version="01.00.04" alias="CoreCA"/>
     <ECCustomAttributes>
-        <ProductionStatus xmlns="CoreCustomAttributes.01.00.03">
+        <ProductionStatus xmlns="CoreCustomAttributes.01.00.04">
             <SupportedUse>NotForProduction</SupportedUse>
         </ProductionStatus>
         <SchemaLayerInfo xmlns="BisCustomAttributes.01.00.00">
             <Value>Application</Value>
         </SchemaLayerInfo>
     </ECCustomAttributes>
+    <ECEnumeration typeName="BaseTurnoutObjectSettingsAspect_BaseTurnoutObjectSettings_BendingMethod_Enum" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Swiss" value="0" displayLabel="Swiss"/>
+        <ECEnumerator name="German" value="1" displayLabel="German"/>
+    </ECEnumeration>
+    <ECEnumeration typeName="BaseTurnoutObjectSettingsAspect_BaseTurnoutObjectSettings_LengthAlong_Enum" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="CenterLine" value="0" displayLabel="CenterLine"/>
+        <ECEnumerator name="Inner" value="1" displayLabel="Inner"/>
+        <ECEnumerator name="Outer" value="2" displayLabel="Outer"/>
+    </ECEnumeration>
+    <ECEnumeration typeName="BaseTurnoutObjectSettingsAspect_BaseTurnoutObjectSettings_NumberOfBranches_Enum" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="One" value="1" displayLabel="1"/>
+        <ECEnumerator name="Two" value="2" displayLabel="2"/>
+    </ECEnumeration>
+    <ECEnumeration typeName="CableAspect_Cable_ShowCable_BoolEnum" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="True" value="1" displayLabel="Yes"/>
+        <ECEnumerator name="False" value="0" displayLabel="No"/>
+    </ECEnumeration>
```

Comment:
This isn't a great pattern, I see this isn't a new pattern (there are many existing enums with this pattern) so I won't insist that it be changed but it should be changed.  Once released it's a breaking change to change the type of the enum even if it is functionally the same, this is because it could break existing ECSql that references the enum

Reply:
This was generated by a tool we use (cif2bis) and is output of a properties service( in civilplatform) written to get all the presentation classes from Civil products. In case we change the pattern manually, next time we run the tool, this will be again updated to the existing pattern. If this needs to handled , can you suggest the pattern that is favorable?